src/constitutive/TACSBladeStiffenedShellConstitutive.cpp

```diff
@@ -1890,21 +1742,32 @@ TacsScalar TACSBladeStiffenedShellConstitutive::computeStiffenerFailure(
   TACSOrthotropicPly* ply = this->stiffenerPly;
 
   // Compute the strain state at the tip of the stiffener
-  TacsScalar zTipOffset = -(this->stiffenerHeight + this->stiffenerThick) -
-                          this->computeStiffenerCentroidHeight();
-  TacsScalar tipStrain[3];
-  memset(tipStrain, 0, 3 * sizeof(TacsScalar));
-  tipStrain[0] = stiffenerStrain[0] + zTipOffset * stiffenerStrain[2];
+  TacsScalar zCentroid = this->computeStiffenerCentroidHeight();
+  TacsScalar zTipOffset =
+      -(this->stiffenerHeight + this->stiffenerThick) - zCentroid;
```

Comment:
Because in this model I assume that the stiffener height value is the length of the stiffener web above the flange, not including the flange.  See below:

![image](https://github.com/smdogroup/tacs/assets/20371123/82598607-8ddf-458a-a939-0d89d6d31412)

I went back and forth on whether to do it this way or to make the stiffener height the total height from the bottom to top of the stiffener. Either way you do it the equations get a bit tricky somewhere. With this approach the calculation of the stiffener cross-section properties is simpler but you have to include the thickness of the flange when computing offsets like this. If you do it the other way then computing the stiffener cross section properties is more complicated because you have to avoid double counting the material that's in both the flange and the web.